signalfx/util.go

```diff
@@ -50,6 +50,22 @@ var ChartColorsSlice = []chartColor{
 	{"lime_green", "#6bd37e"},
 }
 
+// getPayload for viz objects is called withs: *schema.ResourceDiff or *schema.ResourceData - that's why we need this interface
+type ResourceDataAccess interface {
+	Get(key string) interface{}
+	GetOk(key string) (interface{}, bool)
+}
+
+func chartValidate(ctx context.Context, chartResource *schema.ResourceDiff, config interface{}, getPayload func(ResourceDataAccess) (*chart.CreateUpdateChartRequest, error)) error {
+	payload, err := getPayload(chartResource)
+
+	if err != nil {
+		return err
+	}
+
+	return config.(*signalfxConfig).Client.ValidateChart(ctx, payload)
```

Comment:
One thing to be conscious of is that if the type isn't the expected value, this code will panic, not expecting that it will but if it were to, it is a terrible user experience. 

I would suggest looking at type checking here.